dcm2niix_gear/dcm2niix/arrange.py

```diff
@@ -138,28 +250,64 @@ def extract_archive_contents(archive_obj, work_dir):
         dcm2niix_input_dir, dirname = setup_dcm2niix_input_dir(filename, work_dir)
         archive_obj.extractall(dcm2niix_input_dir)
 
-    elif len(subdirs) == 1:
+    elif len(subdirs) >= 1:
 
         # Subdirectory name will be used as the dcm2niix input directory name
+        print(f"subdirs: {subdirs}")
         dcm2niix_input_dir, dirname = setup_dcm2niix_input_dir(subdirs[0], work_dir)
+        print(f"dcm2niix_input_dir: {dcm2niix_input_dir}")
+
+        # set temp directory to extract to, before flattening output contents
+        dcm2niix_input_dir_o = dcm2niix_input_dir + "_o"
 
+        # clean slate
+        if os.path.exists(dcm2niix_input_dir):
+            shutil.rmtree(dcm2niix_input_dir)
+        if os.path.exists(dcm2niix_input_dir_o):
+            shutil.rmtree(dcm2niix_input_dir_o)
+
+        # site for initial extraction
+        os.mkdir(dcm2niix_input_dir_o)
+
+        # extract
         if type(archive_obj) == zipfile.ZipFile:
-            archive_obj.extractall(
-                dcm2niix_input_dir, strip_prefix_ziparchive(archive_obj, subdirs[0])
-            )
+            for subdir in subdirs:
+                archive_obj.extractall(
+                    dcm2niix_input_dir_o, strip_prefix_ziparchive(archive_obj, subdir)
+                )
+        elif type(archive_obj) == tarfile.TarFile:
+            for subdir in subdirs:
+                archive_obj.extractall(
+                    dcm2niix_input_dir_o, strip_prefix_tararchive(archive_obj, subdir)
+                )
 
-        if type(archive_obj) == tarfile.TarFile:
-            archive_obj.extractall(
-                dcm2niix_input_dir, strip_prefix_tararchive(archive_obj, subdirs[0])
-            )
+        # flattening: take file leaves in dcm2niix_input_dir_o and move them dcm2niix_input_dir
+        file_set_source, \
+        file_tree_source, \
+        file_name_path_source, \
+        file_set_target, \
+        file_tree_target, \
+        file_name_path_dict_target = flatten_directory(dcm2niix_input_dir_o, dcm2niix_input_dir)
+
+        # clean up
+        shutil.rmtree(dcm2niix_input_dir_o)
+
+        # display before/after
+        file_tree_source_str = '\n'.join(file_tree_source)
+        file_tree_target_str = '\n'.join(file_tree_target)
+
+        log.info(
+            f"\n\nfile_tree_source:\n{file_tree_source_str}\n"
+            f"\nfile_tree_target:\n{file_tree_target_str}\n\n"
+        )
```

Comment:
Can we move the logging to `flatten_directory`? It does not look like flatten_directory actually needs to return anything.

Reply:
👍